Extensions/Firebase/firebasetools.js

```diff
@@ -0,0 +1,99 @@
+/**
+ * Firebase Tools
+ * @author arthuro555
+ */
+
+/*    Initialization of the firebase SDK    */
+
+/** A Timeout function is used to initialize firebase as soon as the project data has been loaded */
+let gdjsWatcher = function() {
+    if (gdjs.projectData === undefined) {
+        setTimeout(gdjsWatcher, 1);
+        return;
+    }
+    let firebaseConfig;
+    try { 
+        firebaseConfig = JSON.parse(gdjs.projectData.properties.firebaseConfig);
+    } catch {
+        console.error("The Firebase configuration is invalid!");
+    }
+    firebase.initializeApp(firebaseConfig);
+};
+gdjsWatcher();
+
+/*    Creation of all Namespaces    */
+
+/**
+ * Firebase Event Tools
+ * @namespace
+ */
+gdjs.evtTools.firebase = {};
+
+/**
+ * Firebase Analytics Tools
+ * @namespace
+ */
+gdjs.evtTools.firebase.analytics = {};
+
+/*    Function Definitions    */
+
+/**
+ * Enables Firebase Analytics and cache the instance to runtimeGame.
+ * @param {gdjs.RuntimeScene} runtimeScene The current {@link gdjs.RuntimeScene} instance.
+ */
+gdjs.evtTools.firebase.analytics.enable = function(runtimeScene) {
+    runtimeScene.getGame()._analyticsInstance = runtimeScene.getGame()._analyticsInstance || firebase.analytics();
```

Comment:
I am not sure if it is faster to access the instance through this "cache" and I find that kinda unclean.

Reply:
> I am not sure if it is faster 

I would say to avoid premature optimisation if it's at the cost of unclean code. Consider runtimeGame as something that should not be touched (if someone does the same for another extension with analytics - boom: either your extension or the other is now broken).

Instead:
- just use firebase.analytics()
- if you prefer to keep a reference, store it in your extension: `gdjs.evtTools.firebase.analytics._analyticsInstance`.

I would suggest to see if you can initialize this at the beginning of the first scene, so that latter you can just do:
`var analyticsInstance =  gdjs.evtTools.firebase.analytics._analyticsInstance;`
and use `analyticsInstance` in your function (so that it's both short and efficient).